topostats/filters.py

```diff
@@ -296,7 +297,7 @@ def gaussian_filter(self, image: np.ndarray, **kwargs) -> np.array:
             **kwargs,
         )
 
-    def filter_image(self) -> None:
+    def filter_image(self, scars_config: dict) -> None:
```

Comment:
If `scars_config` is moved and nested under `filter_config` in `default_config` then there will be no need to pass this as an option here, rather the `def __init__(self)` will need updating to take the argument, which will be a keyword to `**filter_config` and as I've hopefully correctly suggested in the below comment the logic of whether to run scarring is then used before making the call to the scarring method itself.